### PR TITLE
Handle potential undefined value returned from process dictionary

### PR DESCRIPTION
**Pull request by lukebakken**
CI failures such as this suggest that file handle values can be deleted
mid-operation:

https://github.com/rabbitmq/rabbitmq-server/actions/runs/304216635

### CODE REVIEW

**Review by michaelklishin — APPROVED** [review]

File: src/file_handle_cache.erl [hunk 1]
```diff
@@ -449,7 +449,8 @@ needs_sync(Ref) ->
     %% This must *not* use with_handles/2; see bug 25052
     case get({Ref, fhc_handle}) of
         #handle { is_dirty = false, write_buffer = [] } -> false;
-        #handle {}                                      -> true
+        #handle {}                                      -> true;
+        undefined                                       -> true
```
Comment (dumbbell) [inline src/file_handle_cache.erl:453]: Why should the function return `true` if the handle is unknown?

Comment (lukebakken) [reply src/file_handle_cache.erl:453]: Sorry, meant to open this as a draft. It's not done!

**Comment by lukebakken** [comment]
Thanks for the initial pass @dumbbell. After seeing how the value is used I'm pretty sure `false` is the correct return value. I'm running the `rabbitmq-server` full test suite locally and there are no failures yet.

**Comment by dumbbell** [comment]
I tried the `ct-fast` selection of testsuites with the initial patch locally and it went fine. If the function returned `true`, I guess it wouldn't break RabbitMQ, it might be less optimized, but that's about it.

**Review by dumbbell — APPROVED** [review]

**Comment by lukebakken** [comment]
Backport to-do added [here](https://3.basecamp.com/4467044/buckets/19171848/todosets/3106218954).

**Comment by michaelklishin** [comment]
Backported to `v3.8.x`.

**Comment by michaelklishin** [comment]
#421 was backported to `v3.8.x`.

**Pull request merged**